Purchases/Public/RCPurchases.m

```diff
@@ -347,6 +344,17 @@ - (instancetype)initWithAppUserID:(nullable NSString *)appUserID
     return self;
 }
 
+#if TARGET_OS_IOS || TARGET_OS_TV
+#define APP_DID_BECOME_ACTIVE_NOTIFICATION_NAME UIApplicationDidBecomeActiveNotification
+#define APP_WILL_RESIGN_ACTIVE_NOTIFICATION_NAME UIApplicationWillResignActiveNotification
+#elif TARGET_OS_OSX
+#define APP_DID_BECOME_ACTIVE_NOTIFICATION_NAME NSApplicationDidBecomeActiveNotification
+#define APP_WILL_RESIGN_ACTIVE_NOTIFICATION_NAME NSApplicationWillResignActiveNotification
+#elif TARGET_OS_WATCH
+#define APP_DID_BECOME_ACTIVE_NOTIFICATION_NAME NSExtensionHostDidBecomeActiveNotification
+#define APP_WILL_RESIGN_ACTIVE_NOTIFICATION_NAME NSExtensionHostWillResignActiveNotification
+#endif
```

Comment:
🤔 
we'd still be able to define these, right? 

i.e.: would something like this work? 

```swift
import Foundation
#if os(iOS)
import UIKit
#elseif os(macOS)
import NSApplication
#elseif os(watchOS)
import UIKit
#endif

// todo: make internal
@objc(RCCrossPlatformSupport) public class CrossPlatformSupport: NSObject {
#if os(iOS)
    @objc public static let appDidBecomeActiveNotificationName = UIApplication.didBecomeActiveNotification
#elseif os(macOS)
    @objc public static let appDidBecomeActiveNotificationName = NSApplication.didBecomeActiveNotification
#elseif os(watchOS)
    @objc public static let appDidBecomeActiveNotificationName = NSExtensionHost.didBecomeActiveNotification
#endif
}
```

usage: 

```objective-c
[self.notificationCenter addObserver:self
                            selector:@selector(applicationDidBecomeActive:)
                                name:RCCrossPlatformSupport.appDidBecomeActiveNotificationName object:nil];

```


That way `RCPurchases` doesn't have to own the responsibility of knowing which notification is the correct one per platform.

Reply:
it would work for some of the stuff in `RCCrossPlatformSupport`, but not all (for example, I don't think we'd be able to use this approach for importing the correct framework)

Reply:
just read the rest of the code, it seems like this approach would fall pretty much in line with everything else done in SystemInfo

Reply:
HAHahahahahahahahaha you'd think, and I thought so to.
BUT that's not the case at all. I didn't dig into too much and opted to move the `#if` out of the file into the places where they were used since they only applied to 1 location for each `#define.`
Turns out there are subtle differences when you do defines in objc vs Swift and crossing the objc/swift barrier. 
Once we get to 100% swift, we can go back and clean up any weird stuff like this. Right now, our focus is move this over to swift while maintaining functionality during the whole process. Once we're done, there will be plenty of opportunities to make it more swifty and cleaner 😄

Reply:
yeah, the `#define` is not a thing in Swift, and `#if` checks are certainly less flexible, so it wouldn't work stuff like `UI_DEVICE_AVAILABLE`, but why wouldn't the approach above work for `appDidBecomeActiveNotificationName`? Isn't it the same thing we're doing in `platformHeaderConstant`? 

I think the distinction here is that we can't use it to define anything that's pre-processor, but we should be able to use it for regular variables, like in this case, a string

Reply:
Yeah, that would work, but since we're also migrating that file, I felt that putting those defines in the only file that uses it was the right course of action for now. I'm trying to change as little as possible. When we migrate that file, we'll want to switch to what you're suggesting. At this point, I should note, the constants file `RCCrossPlatformSupport` (`CrossPlatformSupport`) has been removed in favor of moving the `#define`s and `#if`s to the location where they are used. We can definitely revisit it after migration, it was just much easier for me to deal with the dependencies this way.